modules/dnn/test/test_model.cpp

```diff
@@ -806,6 +807,85 @@ TEST_P(Test_Model, TextDetectionByEAST)
     );
 }
 
+TEST_P(Test_Model, YOLOv7)
+{
+    std::string weightPath = _tf("onnx/models/yolov7_not_simplified.onnx");
+    std::string imgPath = _tf("dog_orig_size.jpg");
+
+    Size targetSize{640, 640};
+    float conf_threshold = 0.3;
+    float iou_threshold = 0.5;
+
+    // Reference, which is collected with input size of 640x640
+    std::vector<int> refClassIds{1, 16, 7};
+    std::vector<float> refScores{0.9624768f, 0.95940655f, 0.8693003f};
+    // [x1, y1, x2, y2] x 3
+    std::vector<Rect2d> refBoxes{Rect2d(105.99881f, 150.2002f,  472.52933f, 466.508f),
+                                  Rect2d(109.99532f, 246.13585f, 259.82266f, 600.7827f),
+                                  Rect2d(386.13947f, 83.04132f,  575.3909f,  189.83206f)};
+
+    Mat img = imread(imgPath);
```

Comment:
Can `testDetectModel` be used here?

Reply:
Let me have a try.

Reply:
I just recalled a reason why I did not use `testDetectModel`. For YOLOv7, the final confidence for a box is calculated by multiplying the class confidence and the object confidence in line 855 (`conf *= obj_conf;`). I dont see this in `testDetectModel.detect()`, so I did not use `testDetectModel`.